x-pack/legacy/plugins/siem/public/components/inspect/index.test.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount } from 'enzyme';
+import * as React from 'react';
+
+import {
+  TestProviderWithoutDragAndDrop,
+  mockGlobalState,
+  apolloClientObservable,
+} from '../../mock';
+import { createStore, State } from '../../store';
+import { UpdateQueryParams, upsertQuery } from '../../store/inputs/helpers';
+
+import { InspectButton } from '.';
+import { cloneDeep } from 'lodash/fp';
+
+describe('Inspect Button', () => {
+  const refetch = jest.fn();
+  const state: State = mockGlobalState;
+  const newQuery: UpdateQueryParams = {
+    inputId: 'global',
+    id: 'myQuery',
+    inspect: null,
+    loading: false,
+    refetch,
+    state: state.inputs,
+  };
+
+  let store = createStore(state, apolloClientObservable);
+
+  describe('Render', () => {
+    beforeEach(() => {
+      const myState = cloneDeep(state);
+      myState.inputs = upsertQuery(newQuery);
+      store = createStore(myState, apolloClientObservable);
+    });
+    test('Eui Empty Button', () => {
+      const wrapper = mount(
+        <TestProviderWithoutDragAndDrop store={store}>
+          <InspectButton queryId={newQuery.id} inputId="timeline" show={true} title="My title" />
+        </TestProviderWithoutDragAndDrop>
+      );
+      expect(
+        wrapper
+          .find('button[data-test-subj="inspect-empty-button"]')
+          .first()
+          .exists()
+      ).toBe(true);
+    });
+
+    test('Eui Icon Button', () => {
+      const wrapper = mount(
+        <TestProviderWithoutDragAndDrop store={store}>
+          <InspectButton queryId={newQuery.id} show={true} title="My title" />
+        </TestProviderWithoutDragAndDrop>
+      );
+      expect(
+        wrapper
+          .find('button[data-test-subj="inspect-icon-button"]')
+          .first()
+          .exists()
+      ).toBe(true);
+    });
+
+    test('Eui Empty Button disabled', () => {
+      const wrapper = mount(
+        <TestProviderWithoutDragAndDrop store={store}>
+          <InspectButton isDisabled={true} queryId={newQuery.id} show={true} title="My title" />
+        </TestProviderWithoutDragAndDrop>
+      );
+      expect(wrapper.find('.euiButtonIcon').get(0).props.disabled).toBe(true);
```

Comment:
optional nit / FYI: when chaining off the first matching element, `wrapper.find('.euiButtonIcon').first()` may be used in place of `wrapper.find('.euiButtonIcon').get(0)`

Reply:
if I do first(), I can not get the attribute `disabled`

Reply:
All good, and optional anyway, but FWIW,`first()` appears to work for me:

```
    test('Eui Empty Button disabled', () => {
      const wrapper = mount(
        <TestProviderWithoutDragAndDrop store={store}>
          <InspectButton isDisabled={true} queryId={newQuery.id} show={true} title="My title" />
        </TestProviderWithoutDragAndDrop>
      );
      expect(
        wrapper
          .find('.euiButtonIcon')
          .first()
          .props().disabled
      ).toBe(true);
    });
```

```
 PASS  legacy/plugins/siem/public/components/inspect/index.test.tsx
  Inspect Button
    Render
      ✓ Eui Empty Button (64ms)
      ✓ Eui Icon Button (7ms)
      ✓ Eui Empty Button disabled (5ms)
      ✓ Eui Icon Button disabled (5ms)
    Modal Inspect - happy path
      ✓ Open Inspect Modal (218ms)
      ✓ Close Inspect Modal (181ms)
      ✓ Do not Open Inspect Modal if it is loading (6ms)

Test Suites: 1 passed, 1 total
Tests:       7 passed, 7 total
Snapshots:   0 total
Time:        4.98s, estimated 9s
```